src/openloco/date.cpp

```diff
@@ -0,0 +1,107 @@
+#include "interop/interop.hpp"
+#include "date.h"
+
+using namespace openloco::interop;
+
+namespace openloco
+{
+    static loco_global<uint32_t, 0x00525E2C> _current_day;
+    static loco_global<uint16_t, 0x00525E30> _day_counter;
+    static loco_global<int16_t, 0x00525E32> _current_year;
+    static loco_global<int8_t, 0x00525E34> _current_month;
+    static loco_global<int8_t, 0x00525E35> _current_day_of_month;
+    static loco_global<int32_t, 0x0112C810> _current_day_in_olympiad;
+
+    uint32_t current_day()
+    {
+        return _current_day;
+    }
+
+    date current_date()
+    {
+        return date(_current_year, (month_id)(*_current_month), _current_day_of_month);
+    }
+
+    void set_date(const date &date)
+    {
+        _current_day_of_month = date.day;
+        _current_month = (int8_t)date.month;
+        _current_year = date.year;
+    }
+
+    bool update_day_counter()
+    {
+        bool result = false;
+        constexpr uint16_t increment = 682; // ~17s
+
+        // Check if counter is going to wrap
+        if (_day_counter + increment > std::numeric_limits<uint16_t>::max())
+        {
+            _current_day++;
+            result = true;
+        }
+
+        _day_counter += increment;
+        return result;
+    }
+
+    // 0x00495C65
+    // eax: totalDays
+    // returns:
+    //    eax: year
+    //    ebx: month
+    //    edx: day
+    date calc_date(uint32_t totalDays)
+    {
+        constexpr auto base_year = 1800;
+        constexpr auto days_in_year = 365;
+        constexpr auto days_in_olympiad = (365 * 4) + 1;
+        constexpr auto feb_29 = 31 + 28;
+
+        static const int16_t month_table[] =
```

Comment:
constexpr

Reply:
What is the benefit of this as it doesn't make a difference to the generated code?